engine/engine_api_test.go

```diff
@@ -30,6 +31,21 @@ func (e *Engine) LinkSystem() *ipld.LinkSystem {
 	return &e.lsys
 }
 
+// Returns the engine's default provider ID, exposed for testing purposes only.
+func (e *Engine) ProviderID() string {
+	return e.provider.ID.String()
+}
+
+// Returns the engine's default provider addresses, exposed for testing purposes only.
+func (e *Engine) ProviderAddrs() []string {
+	return e.retrievalAddrsAsString()
+}
+
+// Returns the engine's datastore, exposed for testing purposes only.
+func (e *Engine) Datastore() datastore.Datastore {
+	return e.ds
+}
```

Comment:
The tests are already in `engine` package which means they have access to all the engine fields including unexpored ones. 

I don't we need these.

Reply:
Well that's the thing - it wouldn't allow to just do engine.ds in the tests. I've tried before adding these methods.